contrib/babelfishpg_common/src/varbinary.c

```diff
@@ -636,20 +641,36 @@ varcharvarbinary(PG_FUNCTION_ARGS)
 						"varbinary is not allowed. Use the CONVERT function "
 						"to run this query.")));
 
-	/* If typmod is -1 (or invalid), use the actual length */
+	PG_TRY();
+	{
+		collInfo = lookup_collation_table(get_server_collation_oid_internal(false));
+		encoded_data = encoding_conv_util(data, len, PG_UTF8, collInfo.enc, &encodedByteLen);
+	}
+	PG_CATCH();
+	{
+		ereport(ERROR,
+			   (errcode(ERRCODE_INTERNAL_ERROR),
+				errmsg("Failed to convert from data type varchar to varbinary")));
```

Comment:
Could happens when NULL byte exists in input varchar. Cannot think of any test case for this right now.